metricbeat/helper/prometheus/prometheus.go

```diff
@@ -35,7 +35,11 @@ import (
 	"github.com/elastic/beats/v7/metricbeat/mb"
 )
 
-const acceptHeader = `application/openmetrics-text; version=0.0.1,text/plain;version=0.0.4;q=0.5,*/*;q=0.1`
+const (
+	promTextHeader    = `text/plain;version=0.0.4;q=0.5,*/*;q=0.1`
+	openmetricsHeader = `application/openmetrics-text; version=0.0.1,text/plain;version=0.0.4;q=0.5,*/*;q=0.1`
+	acceptHeader      = promTextHeader
```

Comment:
Metricbeat has two collectors, one for [prometheus](https://github.com/elastic/beats/blob/e40b421b254b5c04b64e3ecc671ad36da3786f01/x-pack/metricbeat/module/prometheus/collector/collector.go#L14), and another one for [openmetrics](https://github.com/elastic/beats/blob/e40b421b254b5c04b64e3ecc671ad36da3786f01/x-pack/metricbeat/module/openmetrics/collector/collector.go#L27), they are mostly the same so far, but I think it would make sense if each collector configured the default headers of their choice.

Reply:
Thanks for the comments. yeah but I think for `application/openmetrics-text` the parser would only work for 'previous' openmetrics format, not the new one, so sending `Accept: application/openmetrics-text` could give target endpoints the chance to respond with new format, while at this moment, `text/plain` prometheus format should still be compatible from their perspective. 

This PR is more like a workaround as being mentioned in the [issue](https://github.com/elastic/beats/issues/24554#issuecomment-800904519), the solution would still be updating the parser as suggested by @exekias .

Reply:
I wonder why `application/openmetrics-text; version=0.0.1` includes the new openmetrics types 🤔 . Shouldn't that change be applied with a new version so as to be backwards compatible?

Reply:
good point. I think somehow that is not happening and hence breaking things.